pkg/resources/apiserver/egressselectorconfigmap.go

```diff
@@ -0,0 +1,46 @@
+/*
+Copyright 2020 The Kubermatic Kubernetes Platform contributors.
+
+Licensed under the Apache License, Version 2.0 (the "License");
+you may not use this file except in compliance with the License.
+You may obtain a copy of the License at
+
+    http://www.apache.org/licenses/LICENSE-2.0
+
+Unless required by applicable law or agreed to in writing, software
+distributed under the License is distributed on an "AS IS" BASIS,
+WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+See the License for the specific language governing permissions and
+limitations under the License.
+*/
+
+package apiserver
+
+import (
+	"k8c.io/kubermatic/v2/pkg/resources"
+	"k8c.io/kubermatic/v2/pkg/resources/reconciling"
+	corev1 "k8s.io/api/core/v1"
+)
+
+// EgressSelectorConfigCreator returns function to create cm that contains egress selection configuration for apiserver
+// to work with konnectivity proxy.
+func EgressSelectorConfigCreator() reconciling.NamedConfigMapCreatorGetter {
+	return func() (string, reconciling.ConfigMapCreator) {
+		return resources.KonnectivityKubeApiserverEgress, func(c *corev1.ConfigMap) (*corev1.ConfigMap, error) {
+			c.Data = map[string]string{
+				"egress-selector-configuration.yaml": `
+apiVersion: apiserver.k8s.io/v1beta1
+kind: EgressSelectorConfiguration
```

Comment:
could you build EgressSelectorConfiguration struct instead of putting here a string? 
That would be better for the struct validation. 

https://github.com/kubernetes/apiserver/blob/0ff020160db9f12f71f1278528cb974c978ba750/pkg/apis/apiserver/types.go#L55